### ISSUE TITLE
Geometry degradation when using Colmap 3.8

### ISSUE THREAD

**Issue by MaroonAmor** [nerfstudio-project/nerfstudio#1479]
Dear authors,

@tancik In the latest version of the docker image, Colmap version has been changed to 3.8 in this [PR](https://github.com/nerfstudio-project/nerfstudio/pull/1448). @Zunhammer I did some sanity checks and found that the geometry result (from `ns-export poisson`) cannot be reproduced when I run `nerfacto` on the Colmap estimated data. Here are the details of my sanity checks:
1. Run Nerfstudio v0.1.16 on data estimated by Colmap 3.7 in the docker: the 3D mesh looks good.
2. Run Nerfstudio v0.1.18 on data estimated by Colmap 3.8 in the docker: the 3D mesh surface is worse than that in experiment 1.
3. Run Nerfstudio v0.1.18 on data previously preprocessed by Nerfstudio v0.1.16: the 3D mesh quality can be reproduced as in experiment 1, which rules out the potential cause from the model code of `nerfacto`.
4. Run Nerfstudio v0.1.18 on data estimated by Colmap3.7 (rebuild the docker to change Colmap from 3.8 to 3.7): the mesh quality can be also reproduced as in experiment 1, which means the geometry degradation issue comes from the Colmap 3.8.

Here, I used the same command to run Colmap on the same data, and the same command to run `nerfacto`. It seems like the issue comes from the latest version of Colmap 3.8. The camera pose estimation is not comparable to 3.7.

If that is the case, could we change it back to Colmap 3.7? Please double-check and let me know if I missed anything. Thanks.

```
# Install colmap.
RUN git clone --branch 3.7 https://github.com/colmap/colmap.git --single-branch && \
    cd colmap && \
    mkdir build && \
    cd build && \
    cmake .. -DCUDA_ENABLED=ON \
        	 -DCUDA_NVCC_FLAGS="--std c++14" && \
    make -j && \
    make install && \
    cd ../.. && \
    rm -rf colmap
```

**Comment by Zunhammer** [comment]
I need to check on this. If 3.8 is the issue I'll definitely revert to 3.7. Till now I didn't notice any probelms however I normally don't use the mesh export. But I would expect that we could also see a difference in image quality when rendering a video...

If enyone else can confirm issues with colmap 3.8 please comment here. thanks

**Comment by MaroonAmor** [comment]
@Zunhammer @tancik Yeah, it may not be obvious from the image rendering quality. But I did observe there is gap in the estimated camera intrinsic between Colmap 3.7 and Colmap 3.8. Camer intrinsics from experiment 1 and experiment 4 are numerically close (almost the same). Sorry I cannot share the data but here is the command I was using to run `ns-process-data`

```
ns-process-data images --data path_to_your_original_data --output-dir path_to_your_output_folder --matching-method exhaustive
```

It would be great to have sample data for sanity checks in a building pipeline in CI/CD. This could also be beneficial to image rendering quality.

**Comment by tancik** [comment]
Can you try running with 3.8 but comment out these two lines to see if it makes a difference
https://github.com/nerfstudio-project/nerfstudio/blob/a3a1b1f667b70cd22124d78337556fec502ceca3/nerfstudio/process_data/colmap_utils.py#L561-L562

**Comment by MaroonAmor** [comment]
Wait before doing another trial. @tancik I think these two lines are there in both v0.1.16 and  v0.1.18 (even before v0.1.16). And it already applied to both Colmap 3.7 and Colmap 3.8. Please help me understand why commenting out these two lines would make a difference. Wouldn't it also change the geometry result on Colmap 3.7?

Please correct me if I misunderstood.

PS. @tancik @Zunhammer This could be a bug in Colmap 3.7 where their versioning is coded as 3.8 in their repo: https://github.com/colmap/colmap/blob/ea40ef9aecf98eec7e14bd6eeedd2a5381c6b46b/CMakeLists.txt#L36
So even though we installed Colmap 3.7, it still shows version 3.8 from CLI (just off-topic). :smile:

**Comment by tancik** [comment]
Yes, sorry. I responded too hastily. Colmap 3.8 must have some other parameter that differs which causes issues with the scene you are using. Are you finding it to be consistently worse across multiple scenes?

**Comment by MaroonAmor** [comment]
@tancik @Zunhammer Sure. Let me find a good public dataset from Nerstudio and reproduce the issue. Will keep you posted.

**Comment by MaroonAmor** [comment]
@tancik @Zunhammer Some updates: I had another trial for the poster dataset from Nerstudio. The geometry quality is comparable between Colmap 3.7 and Colmap 3.8. I didn't observe the degradation issue.

I revised my data with the geometry degradation issue. The issue was due to some challenging frames during the feature-matching step that gave me different estimated frames between Colmap 3.7 and Colmap 3.8: the subset of estimated frames is not exactly the same. For instance, the dataset I have has 88 frames in total. If I ran Colmap 3.7, 86 frames are estimated, while when running Colmap 3.8, only 85 frames are estimated.

So I did another check: I ran Colmap 3.8 on the subset of 86 frames (that Colmap 3.7 is able to estimate), and all 86 frames were also estimated successfully by Colmap 3.8. The geometry quality then was reproducible and comparable to that from Colmap 3.7.

I will close the issue for now, since the best way to address this issue is to make sure we have good image quality during capturing in order to run Colmap successfully on all frames.
